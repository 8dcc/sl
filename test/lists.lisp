;;------------------------------------------------------------------------------
;; Features tested in this source:
;;   - Primitives for creating lists (`list', `append')
;;   - Primitives for accessing lists (`car', `cdr')
;;------------------------------------------------------------------------------

(list)          ; Expected: nil
(list 'a 'b 'c) ; Expected: (a b c)

(cons 'a nil)        ; Expected: (a)
(cons 'a '(b c))     ; Expected: (a b c)
(cons '(a b) '(y z)) ; Expected: ((a b) y z)

(car nil)    ; Expected: nil
(car '(a))   ; Expected: a
(car '(a b)) ; Expected: a

(cdr nil)      ; Expected: nil
(cdr '(a))     ; Expected: nil
(cdr '(a b c)) ; Expected: (b c)

(append)                   ; Expected: nil
(append nil)               ; Expected: nil
(append '(a) '(b c) '(d))  ; Expected: (a b c d)
(append '(a b) nil '(c d)) ; Expected: (a b c d)
