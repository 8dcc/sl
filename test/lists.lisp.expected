Standard library loaded.
nil
(a b c)
nil
nil
(a b c d)
(a b c d)
(a)
(a . b)
(a b c)
(a b c)
((a b) y z)
nil
a
a
a
nil
nil
b
(b c)
0
3
